File: hyperexponential.rater.rex-main/source_files/6350_v1.32.1/view/rate_change.js
import * as HX from "hx-model-components";
import { render_notifications } from "view/results_tables";

function rate_change() {
  return (
    <HX.Page title="Rate Change" fullWidth={true} shownBy="model_state/show_after_landing_page">
      {render_notifications()}
      <HX.Section>
        <HX.Pane flow="right">
          <HX.Collection fields={[
            "rate_change/expiring_policy_option_id",
          ]} />
          <HX.Button task="expiring_policy_fetch_task" title="Fetch Expiring Data" />
        </HX.Pane>
        <HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection fields={[
              // "policy_information/insured_output",
              "rate_change/layer/renewal",
              "rate_change/layer/expiring",
            ]} />
            <HX.Pane />
            <HX.Pane />
          </HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection title="Premium - Policy Term (100% Share)" fields={[
              "renewal",
              "expiring",
            ]} with="rate_change/premium_policy_term_100pct" />
            <HX.Collection title="Premium - Policy Term (Beazley Share)" fields={[
              "renewal",
              "expiring",
            ]} with="rate_change/premium_policy_term" />
            <HX.Pane />
          </HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection title="Premium - Annualized (100% Share)" fields={[
              "renewal",
              "expiring",
            ]} with="rate_change/premium_annualized_100pct" />
            <HX.Collection title="Premium - Annualized (Beazley Share)" fields={[
              "renewal",
              "expiring",
            ]} with="rate_change/premium_annualized" />
            <HX.Pane />
          </HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection title="BPro Premium"
              fields={[
                "rate_change/expiring_bpro_premium",
              ]} />
            <HX.Pane />
            <HX.Pane />
          </HX.Pane>
        </HX.Pane>
      </HX.Section>
      <HX.Section title="UW Rate Change">
        <HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection title="Model %"
              fields={[
                "exposure_change/ratio",
                "risk_characteristics_change/ratio",
                "deductible_change/ratio",
                "limit_change/ratio",
                "terms_and_conditions_change/ratio",
                "other_change/ratio",
              ]} with="rate_change" />
            <HX.Collection title="UW Selected %"
              fields={[
                "exposure_change/ratio_uw",
                "risk_characteristics_change/ratio_uw",
                "deductible_change/ratio_uw",
                "limit_change/ratio_uw",
                "terms_and_conditions_change/ratio_uw",
                "other_change/ratio_uw",
              ]} with="rate_change" />
            <HX.Collection title="Comments"
              fields={[
                "exposure_change/comments",
                "risk_characteristics_change/comments",
                "deductible_change/comments",
                "limit_change/comments",
                "terms_and_conditions_change/comments",
                "other_change/comments",
              ]} with="rate_change" />
            <HX.Collection title="Renewal"
              fields={[
                "exposure_change/renewal",
                "risk_characteristics_change/renewal",
                "deductible_change/renewal",
                "limit_change/renewal",
                "terms_and_conditions_change/renewal",
                "other_change/renewal_brokerage",
                "other_change/renewal_signed_line",
              ]} with="rate_change" />
            <HX.Collection title="Expiring"
              fields={[
                "exposure_change/expiring",
                "risk_characteristics_change/expiring",
                "deductible_change/expiring",
                "limit_change/expiring",
                "terms_and_conditions_change/expiring",
                "other_change/expiring_brokerage",
                "other_change/expiring_signed_line",
              ]} with="rate_change" />
          </HX.Pane>
          <HX.Pane flow="right">
            <HX.Collection title="Rate Changes"
              fields={[
                "total_factor/technical",
                "risk_adjusted_rate_change/technical",
                "simple_rate_change"
              ]} with="rate_change" />
            <HX.Collection title="Underwriter Selected"
              fields={[
                "total_factor/underwriter",
                "risk_adjusted_rate_change/underwriter"
              ]} with="rate_change" />
            <HX.Pane />
          </HX.Pane>
        </HX.Pane>
      </HX.Section>

    </HX.Page >
  )
}

export { rate_change };

File: hyperexponential.rater.rex-main/source_files/6350_v1.32.1/data_schema/rate_change.py
import hx_data_schema as hx
from data_schema.utilities import thousands_format, percent_format

def rate_change():
    # Section Policy Information
    return  {
        "rate_change": hx.Structure(children={
            "expiring_policy_info": hx.Structure(children={
                "expiring_num_locs": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_tiv_total": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_policy_length": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_achieved_premium": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_limit": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_excess": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_deductible": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_brokerage": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_writ_line": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
                "expiring_expected_loss": hx.Float(mode="input", optionality="optional", default=None, async_output=["expiring_policy_fetch_task"], view={"read_only": True}),
            }),
            "expiring_policy_option_id": hx.Int(mode="override", async_input=["expiring_policy_fetch_task"], view={"label": "Expiring Policy Option", "format": {"thousandSeparated": False}}),
            "expiring_bpro_premium": hx.Float(mode="input", default=0, async_output=["expiring_policy_fetch_task"], view={"label": "Expiring", "format": {"thousandSeparated": True, "mantissa": 0}}),
            "layer": hx.Structure(children={
                "renewal": hx.Int(default=1, mode="input", options=[1, 2, 3, 4, 5, 6], async_input=["expiring_policy_fetch_task"], view={"label": "Renewal Layer"}),
                "expiring": hx.Int(default=1, mode="input", options=[1, 2, 3, 4, 5, 6], async_input=["expiring_policy_fetch_task"], view={"label": "Expiring Layer"}),
            }),
            "premium_policy_term": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": "Renewal", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": "Expiring", "format": {"thousandSeparated": True, "mantissa": 0}}),
            }),
            "premium_policy_term_100pct": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": "Renewal", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": "Expiring", "format": {"thousandSeparated": True, "mantissa": 0}}),
            }),
            "premium_annualized": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": "Renewal", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": "Expiring", "format": {"thousandSeparated": True, "mantissa": 0}}),
            }),
            "premium_annualized_100pct": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": "Renewal", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": "Expiring", "format": {"thousandSeparated": True, "mantissa": 0}}),
            }),
            "exposure_change": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "ratio": hx.Float(mode="output", view={"label": "Exposure change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "risk_characteristics_change": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "ratio": hx.Float(mode="output", view={"label": "Risk Characteristic change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "deductible_change": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "ratio": hx.Float(mode="output", view={"label": "Deductible change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "limit_change": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "ratio": hx.Float(mode="output", view={"label": "Limit change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "terms_and_conditions_change": hx.Structure(children={
                "renewal": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "expiring": hx.Float(mode="output", view={"label": " ", "format": {"thousandSeparated": True, "mantissa": 0}}),
                "ratio": hx.Float(mode="output", view={"label": "Terms and conditions change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "other_change": hx.Structure(children={
                "renewal_brokerage": hx.Float(mode="output", view={"label": "Brokerage", "format": percent_format(mantissa=2)}),
                "renewal_signed_line": hx.Float(mode="output", view={"label": "Signed Line", "format": percent_format(mantissa=2)}),
                "expiring_brokerage": hx.Float(mode="output", view={"label": "Brokerage", "format": percent_format(mantissa=2)}),
                "expiring_signed_line": hx.Float(mode="output", view={"label": "Signed Line", "format": percent_format(mantissa=2)}),
                "ratio": hx.Float(mode="output", view={"label": "Other factors change dictates change of", "format": percent_format()}),
                "ratio_uw": hx.Float(mode="override", view={"label": " ", "format": percent_format()}),
                "comments": hx.Str(mode="input", default="", view={"label": " "}),
            }),
            "total_factor": hx.Structure(children={
                "technical": hx.Float(mode="output", view={"label": "Total Factor", "format": percent_format()}),
                "underwriter": hx.Float(mode="output", view={"label": "Total Factor", "format": percent_format()}),
            }),
            "simple_rate_change": hx.Float(mode="output", view={"label": "Simple Rate Change", "format": percent_format()}),
            "risk_adjusted_rate_change": hx.Structure(children={
                "technical": hx.Float(mode="output", view={"label": "Risk Adjusted Rate Change", "format": percent_format()}),
                "underwriter": hx.Float(mode="output", view={"label": "Risk Adjusted Rate Change", "format": percent_format()}),
            }),
            "error_message": hx.Str(mode="output"),
            })
        }
    